README.md

```diff
@@ -1,366 +1,542 @@
+`-recursive` flag followed by the folder path. For example:
 
 ```bash
-./sda-cli -config download <configuration_file> -dataset-id <datasetID> -url <download-service-url> -outdir <outdir> --from-file <path/to/text_file>
+./sda-cli -config <configuration_file> download -dataset-id <datasetID> -url <download-service-url> -outdir <outdir> -recursive <path_to_folder> 
 ```
 
-#### Download all the files of the dataset
+This command preserves the folder structure of the specified directories during
+the download process.
+
+To recursively download multiple folders, list them separated by spaces.
+
+### Download files from a list file
 
-For downloading the whole dataset the user needs add the `--dataset` flag and NOT providing any filepaths:
+To download multiple files, you can also provide a text file containing the file
+paths. Each path should be on a separate line. Use the `-from-file` flag followed
+by the path to the text file, as shown below:
 
 ```bash
-./sda-cli -config download <configuration_file> -dataset-id <datasetID> -url <download-service-url> -outdir <outdir> --dataset
+./sda-cli -config <configuration_file> download -dataset-id <datasetID> -url <download-service-url> -outdir <outdir> -from-file <path_to_list_file>
 ```
 
-where the dataset will be downloaded in the `<outdir>` directory be keeping the original folder structure of the dataset.
+This approach simplifies downloading large numbers of files.
 
-#### Download encrypted files using the download API
+### Download all the files of a dataset
 
-When a [public key](#create-keys) is provided, you can download files that are encrypted on the server-side with that public key. The command is similar to downloading the unencrypted files except that a public key is provided through the `-pubkey` flag. For example:
+To download all files in a dataset, use the `-dataset` flag without providing
+any argument.
 
 ```bash
-./sda-cli -config <configuration_file> download -pubkey <public-key-file> -dataset-id <datasetID> -url <download-service-url> -outdir <outdir> <filepath_1_to_download> <filepath_2_to_download> ...
+./sda-cli -config <configuration_file> download -dataset-id <datasetID> -url <download-service-url> -outdir <outdir> -dataset
 ```
 
-After a successful download, the encrypted files can be [decrypted](#decrypt-file) using the private key corresponding to the provided public key.
+The dataset will be downloaded to the `<outdir>`, preserving its original folder
+structure.
 
-## Decrypt file
+### Download encrypted files
 
-Given that the instructions in the [download section](#download) have been followed, the key pair and the data files should be stored in some location. The last step is to decrypt the files in order to access their content. That can be achieved using the following command:
+When a [public key](#create-crypt4gh-key-pair) is provided, you can download
+files encrypted on the server-side with that key. The syntax is similar to
```

Comment:
Note that only file-by-file downloads are currently supported when downloading encrypted files.

Reply:
Good point! Fixed in 7027802